pkg/portableresources/backend/controller/createorupdateresource.go

```diff
@@ -119,15 +119,25 @@ func (c *CreateOrUpdateResource[P, T]) Run(ctx context.Context, req *ctrl.Reques
 			return ctrl.Result{}, err
 		}
 	}
-
 	if recipeDataModel.Recipe() != nil {
 		recipeDataModel.Recipe().DeploymentStatus = util.Success
 	}
+
+	datamodel := recipeDataModel.(rpv1.RadiusResourceModel)
+
+	// Set recipe details on the resource metadata used.
+	if recipeOutput != nil && recipeOutput.Status != nil {
+		datamodel.ResourceMetadata().Status.Recipe = &rpv1.RecipeStatus{
+			TemplateKind:    recipeOutput.Status.TemplateKind,
+			TemplatePath:    recipeOutput.Status.TemplatePath,
+			TemplateVersion: recipeOutput.Status.TemplateVersion,
```

Comment:
Since this is optional, should a nil check be added?

Reply:
It's a string not a ptr...therefore did not add a nil check